src/Runner.Listener/JobDispatcher.cs

```diff
@@ -712,6 +716,116 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
             }
         }
 
+        // Best effort upload any logs for this job.
+        private async Task TryUploadUnfinishedLogs(Pipelines.AgentJobRequestMessage message)
+        {
+            Trace.Entering();
+
+            var logFolder = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Diag), PagingLogger.PagingFolder);
+            if (!Directory.Exists(logFolder))
+            {
+                return;
+            }
+
+            var logs = Directory.GetFiles(logFolder);
+            if (logs.Length == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection));
```

Comment:
pull the cred from the job message, we will use this cred to talk to actions service and upload the log.